### ISSUE TITLE
Today CreateCoreWebView2EnvironmentWithDetails stopped working

### ISSUE THREAD

**Issue by pontusn** [MicrosoftEdge/WebViewFeedback#151]
Previously functional code stopped working today. CreateCoreWebView2EnvironmentWithDetails does not create environment, callback receives E_NOINTERFACE and NULL as argument.

**Comment by david-risney** [comment]
Thanks for letting us know! Which version of the SDK are you using? Does the loader DLL version match the header version?

**Comment by pontusn** [comment]
We use the latest (0.9.430). Everything match. Our binaries worked fine yesterday, same binary fails today. Feels like something changed in Canary-build.

David Risney <notifications@github.com> skrev: (8 april 2020 19:10:06 CEST)
>Thanks for letting us know! Which version of the SDK are you using?
>Does the loader DLL version match the header version?
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/MicrosoftEdge/WebViewFeedback/issues/151#issuecomment-611079794

-- 
Skickat från min Android-enhet med K-9 Mail. Ursäkta min fåordighet.

**Comment by pontusn** [comment]
Switching from Canary 83.0.478.1 to Dev 83.0.474.0 resolves the issue in my end. My take is that something is broken in the current Canary-build.

**Comment by pontusn** [comment]
Some example code:
```
    HRESULT hRes = CreateCoreWebView2EnvironmentWithDetails(
        nullptr,
        szAppData,
        startArguments.data(),
        Callback<ICoreWebView2CreateCoreWebView2EnvironmentCompletedHandler>(
            [this](HRESULT result, ICoreWebView2Environment* webviewEnvironment) -> HRESULT {
                ATLTRACE(_T("ICoreWebView2CreateCoreWebView2EnvironmentCompletedHandler\n"));
                m_spWebViewEnvironment = webviewEnvironment;
                if (!m_spWebViewEnvironment)
                {
                    // Fails here...
                    m_strStatusMessage.Format(L"Failed to create environment. hResult=0x%08x", result);
                    return result;
                }
                m_spWebViewEnvironment->CreateCoreWebView2Host(m_hWnd, Callback<ICoreWebView2CreateCoreWebView2HostCompletedHandler>(
                    [this](HRESULT result, ICoreWebView2Host* webviewHost) -> HRESULT {
                        ATLTRACE(_T("ICoreWebView2CreateCoreWebView2HostCompletedHandler\n"));
                        m_spWebViewHost = webviewHost;
                        if (!m_spWebViewHost)
                        {
                            m_strStatusMessage.Format(L"Failed to create host. hResult=0x%08x", result);
                            return result;
                        }

                        if (FAILED(result = m_spWebViewHost->get_CoreWebView2(&m_spWebViewWindow)) || !m_spWebViewWindow)
                        {
                            m_strStatusMessage.Format(L"Failed to get view. hResult=0x%08x", result);
                            return FAILED(result) ? result : E_FAIL;
                        }

                        if (FAILED(result = InternalConfigure()))
                        {
                            m_strStatusMessage.Format(L"Failed to configure browser. hResult=0x%08x", result);
                            return result;
                        }

                        return S_OK;
                    }).Get());
                return S_OK;
            }).Get());
    if (FAILED(hRes))
    {
        m_strStatusMessage.Format(L"Failed to start WebView2. hResult=0x%08x", hRes);
    }

```

**Comment by omegakode** [comment]
Same here i update to canary  83.0.478.1 and the code stopped working for the same reason.

**Comment by jspuij** [comment]
Same here.

**Comment by david-risney** [comment]
Thanks for reporting this! Yes, it was a problem in the canary versions you identified. It is fixed in 84.0.480.0. Apologies for this issue and thanks again for reporting it!
